### ISSUE TITLE
Proposal to improve security with HSTS and Public-key-pins

### ISSUE THREAD

**Issue by onewaypub** [owncloud/client#2156]
Since firefox (partially since 32 and full till 34) and chrome support HSTS and HTTP public key extension, I would recommend to implement these feature also in the client. This means:
- HSTS: Client know the server has only secured communication. 
- HTTP public key extension (Public-Key-Pins (PKP)): So use this information to verify if the stored hash is okay with the hash of the given cert and is not changed. If something is wrong show a warning.

**Comment by LukasReschke** [comment]
cc @karlitschek That's what we basically discussed :-)

**Comment by karlitschek** [comment]
@LukasReschke Thanks :-)

**Comment by thedd** [comment]
Since this issue has been open vor a long time: Is there any progress or is anyone working on implementing HPKP for the client?

**Comment by danimo** [comment]
@thedd No. It's simply not high enough on the priority list right now.

**Comment by ogoffart** [comment]
Qt 5.9 has QNetworkAccessManager::setStrictTransportSecurityEnabled we could use.

But since we store the redirected URL in the settings, i don't think it is necessary.
